src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1473,6 +1473,20 @@ protected override void InvokeDefaultAction(string path)
             return true;
         }
 
+        private bool DetectRecursiveSymLink(DirectoryInfo path)
+        {
+            var readSymLink = Platform.NonWindowsInternalGetTarget(path.FullName);
+
+            if (path.FullName.Contains(readSymLink))
+            {
+                Console.WriteLine("\n Detected recursive symlink pointing to its parent directory.");
+                Console.WriteLine(" Skipping: " + path.FullName + " ==> " + readSymLink);
```

Comment:
We avoid `Console.Write*` for any sort of diagnostics, instead preferring the host apis. A host might not have a console at all but still want to show (e.g. log) messages.